src/parser/deserializer.ts

```diff
@@ -670,14 +764,25 @@ function isolateEval(
   return functionCache[functionString].bind(object);
 }
 
-function getValidatedString(buffer: Buffer, start: number, end: number) {
+function getValidatedString(
+  buffer: Buffer,
+  start: number,
+  end: number,
+  validation: Document,
```

Comment:
I think this method should only care about whether or not to validateUtf8, so both of these can be condensed to a single `validateUtf8: boolean`; then lines 776 and 777 get condensed to `if (validateUtf8)`